libsyclinterface/source/dpctl_sycl_extension_interface.cpp

```diff
@@ -62,3 +62,33 @@ bool DPCTLWorkGroupMemory_Available()
     return false;
 #endif
 }
+
+DPCTL_API
+__dpctl_give DPCTLSyclRawKernelArgRef DPCTLRawKernelArg_Create(void *bytes,
+                                                               size_t count)
+{
+    DPCTLSyclRawKernelArgRef rka = nullptr;
+    try {
+        auto RawKernelArg = new RawKernelArgData{bytes, count};
```

Comment:
Could you please use `unique_ptr` here?
```C++
auto RawKernelArg = std::make_unique(new RawKernelArgData{bytes, count});
rka = wrap<RawKernelArgData>(RawKernelArg.release());
```

Reply:
I'm not sure I understand the advantage here. We're still allocating with `new` and we're immediately releasing the `unique_ptr` in the second line. Is there any benefit of this, e.g., exception handling?

Reply:
It also deviates from the code style used in the rest of `libsyclinterface`, e.g., [here](https://github.com/IntelPython/dpctl/blob/master/libsyclinterface/source/dpctl_sycl_queue_interface.cpp#L326).

Reply:
Yes, biggest advantage would be not having to delete it in case an exception is caught.

If an exception did occur here, we would need `delete rka` below in the catch block.

Reply:
Using `unique_ptr` know.